README.md

```diff
@@ -1,27 +1,143 @@
+
+The utilities can be expanded further. Let's look at the credential login process now and go over what each step is doing and where you can take the code and add it to your existing project or start from scratch from this point.
 
 ## Disclaimer
-This code is not for production use. It serves as an example workflow for accepting Credentials. 
+
+This code is not for production use. It serves as an example workflow for accepting Credentials.
 
 ## Getting Started
 
-First, install dependencies:
+Lets get going! Lets see how you can start a web3 DID login with credentials verification.
+
+### Requirements
+
+`Node` installed and [sporran wallet](https://github.com/BTE-Trusted-Entity/sporran-extension/tree/main) for testing use the test sporran wallet. For installation follow the steps in the sporran wallet repository.
+
+
+### Setup
+
+Clone the repository and go into the folder or you can setup a whole project from start to finish following the  [KILT Distillery CLI](https://github.com/KILTprotocol/kilt-distillery-cli/).
+
+In order to set up the application several environment variables and files must be set up manually or through the Distillery:
+
+- .env
+- didConfiguration.json
+
+### Environment variables
+
+The following variables must be generated. This can be done via the Distillery or manually.
+
+```js
+DAPP_NAME='NAME OF APPLICATION'
+JWT_SECRET='JSON WEB TOKEN SECRET HERE'
+JWT_EXPIRY='TIME WITH DEMONINATION E.G. Seconds (s) or Hours (h) or days (d)'
+JWT_RENEW='TRUE or FALSE'
+ORIGIN='ORGIN OF THE APPLCIATION'
+WSS_ADDRESS='CONNECT TO CHAIN'
+VERIFIER_MNEMONIC="Enter your twelve word mnemonic phrase here "
+VERIFIER_ADDRESS='Enter your address'
+VERIFIER_DID_URI='Enter your DID URI'
+```
+
+#### Blockchain address
+
+Before you call any SDK functionality, you need to initialize the crypto libraries and configure the SDK. For the project, we'll be using Peregrine Testnet. This is done by calling await Kilt.init({ address }) where address is the address of the full node you want to connect to. For the project, use `wss://peregrine.kilt.io/parachain-public-ws`. Add the address to your `.env` file as `WSS_ADDRESS`.
+
+#### JSON Web token
+
+Covering the JSON web token, known as JWT, management and creation is out of the scope of the project.
+
+#### Verifier Setup
+
+In order to use the web3 login, the service needs its own DID and provide a [well known domain linkage credential](https://identity.foundation/specs/did-configuration/). A well known domain linkage credential provides a credential for users to verify the domain belongs to the given service. Creating a way to share public keys with the application and user. The DID configuration is used to prevent replay attacks and man in the middle attacks, more details can be found in the [credential Api specification](https://github.com/KILTprotocol/credential-api#man-in-the-middle).
```

Comment:
```suggestion
In order to use the web3 login, the service needs its own DID and must provide a [well-known domain linkage credential](https://identity.foundation/specs/did-configuration/). A well-known domain linkage credential provides a credential for users to verify the domain belongs to the given service. Creating a way to share public keys with the application and user. The DID configuration is used to prevent replay attacks and man-in-the-middle attacks, more details can be found in the [credential Api specification](https://github.com/KILTprotocol/credential-api#man-in-the-middle).
```